FSE-193: Add the option for where log info is sent for process.

tduffield opened the pull request:
You can either write them to the standard /var/log file using `file` or print them to STDOUT using `stdout`. You can set which one you want using a node attribute (the same way you set the runit commnad). 

To handle this I created two classes: Log::Reader and Log::Writer that will handle the transport of log lines from runit to chef-init and finally to STDOUT.

Commit:
Add support for multiple logs types

- User can specify whether logs should be written to /var/log or STDOUT
- Cleanup RSpec tests

Commit:
Add Log::Reader and Log::Writer classes

Add two classes: Log::Reader and Log::Writer that will handle the transport of log lines from runit to chef-init and finally to STDOUT.

Commit:
Update CHANGELOG

stevendanna commented on lib/chef-init/log/reader.rb:
```diff
@@ -0,0 +1,75 @@
+      end
+
+      def create_pipe
+        system_command("mkfifo #{@pipe}")
```
I believe that this will fail if the file already exists.  Not sure if it makes sense to handle that case because of how this tool is used though.

Commit:
Replace Log classes with Loggers classes

To simplify things, avoid interleaved messages and take advantage of runit's default behavior I am replacing ChefInit::Log::Reader and ChefInit::Log::Writer with ChefInit::Loggers. These are loggers that will read from STDIN and send the logs to various destinations. Right now, the only solution we have is for writing to STDOUT.

Commit:
Bump version to 0.4.0.dev

Commit:
Update dep on Chef; rely on Chef for mixlib deps

alambike commented:
Is this going to be merged? I was trying to use chef-init with chef 12.0.3 and https://github.com/opscode/chef-init/commit/39ea2570c51da44c19574855eff121cc45155a55 solve the dependencies problem:

```
# /opt/chef/embedded/bin/chef-init --verify
[2014-12-20T21:32:16+00:00] INFO: file '/opt/chef/embedded/bin/runsvdir' exists: pass
[2014-12-20T21:32:16+00:00] INFO: file '/opt/chef/embedded/bin/sv' exists: pass
[2014-12-20T21:32:16+00:00] INFO: file '/opt/chef/bin/chef-init' exists: pass
[2014-12-20T21:32:16+00:00] INFO: file '/usr/bin/chef-init' exists: pass
[2014-12-20T21:32:16+00:00] INFO: file '/opt/chef/bin/chef-client' exists: pass
[2014-12-20T21:32:16+00:00] INFO: file '/usr/bin/chef-client' exists: pass
[2014-12-20T21:32:16+00:00] INFO: command `/opt/chef/bin/chef-init --version` runs: pass
[2014-12-20T21:32:16+00:00] INFO: command `/usr/bin/chef-init --version` runs: pass
[2014-12-20T21:32:19+00:00] INFO: command `/opt/chef/bin/chef-client --version` runs: pass
[2014-12-20T21:32:21+00:00] INFO: command `/usr/bin/chef-client --version` runs: pass
[2014-12-20T21:32:44+00:00] INFO: bootstrap: failing chef-client exit codes are honored: pass
[2014-12-20T21:32:44+00:00] INFO: failing bootstrap - runsvdir process is cleaned up: pass
[2014-12-20T21:32:44+00:00] INFO: failing bootstrap - runsv process is cleaned up: pass
[2014-12-20T21:32:44+00:00] INFO: failing bootstrap - chef-init-test process is cleaned up: pass
[2014-12-20T21:33:06+00:00] ERROR: bootstrap: passing chef-client exit codes are honored: fail
[2014-12-20T21:33:06+00:00] ERROR: Expected `0` but received `1`
[2014-12-20T21:33:06+00:00] ERROR: ====== Command Output ======
[2014-12-20T21:33:06+00:00] FATAL: Configuration error Gem::ConflictError: Unable to activate chef-init-0.3.3.5, because mixlib-shellout-2.0.0 conflicts with mixlib-shellout (~> 1.4), chef-12.0.3 conflicts with chef (= 11.12.8)
[2014-12-20T21:33:06+00:00] FATAL:   /tmp/chef20141220-1997-oqx1x9/zero.rb:1:in `from_string'
[2014-12-20T21:33:06+00:00] FATAL: Aborting due to error in '/tmp/chef20141220-1997-oqx1x9/zero.rb'

[2014-12-20T21:33:06+00:00] INFO: successful bootstrap - runsvdir process is cleaned up: pass
[2014-12-20T21:33:06+00:00] INFO: successful bootstrap - runsv process is cleaned up: pass
[2014-12-20T21:33:06+00:00] INFO: successful bootstrap - chef-init-test process is cleaned up: pass
[2014-12-20T21:33:06+00:00] INFO: runsvdir started: pass
[2014-12-20T21:33:06+00:00] INFO: enabled services started: pass
[2014-12-20T21:33:06+00:00] INFO: onboot - runsvdir process is cleaned up: pass
[2014-12-20T21:33:06+00:00] INFO: onboot - runsv process is cleaned up: pass
[2014-12-20T21:33:06+00:00] INFO: onboot - chef-init-test process is cleaned up: pass
[2014-12-20T21:33:06+00:00] FATAL: Tests failed:
bootstrap: chef-init does not honor passing exit code from chef-client

```
